references/cli/ls_test.go

```diff
@@ -91,26 +144,38 @@ func TestBuildApplicationListTable(t *testing.T) {
 		t.Run(name, func(t *testing.T) {
 			r := require.New(t)
 
-			if tc.app != nil {
-				err := client.Create(ctx, tc.app)
-				r.NoError(err)
+			if tc.apps != nil && len(tc.apps) > 0 {
+				for _, app := range tc.apps {
+					err := client.Create(ctx, app)
+					r.NoError(err)
+				}
 			}
 			service := map[string]common.ApplicationComponentStatus{}
-			for _, s := range tc.app.Status.Services {
-				service[s.Name] = s
+			for _, app := range tc.apps {
+				for _, s := range app.Status.Services {
+					service[s.Name] = s
+				}
 			}
 
-			tb, err := buildApplicationListTable(ctx, client, "test")
-			r.Equal(tc.expectedErr, err)
+			LabelSelector = tc.labelSelector
 
-			for i, component := range tc.app.Spec.Components {
-				row := tb.Rows[i+1]
-				compName := fmt.Sprintf("%s", row.Cells[1].Data)
-				r.Equal(component.Name, compName)
-				r.Equal(component.Type, fmt.Sprintf("%s", row.Cells[2].Data))
-				r.Equal(string(tc.app.Status.Phase), fmt.Sprintf("%s", row.Cells[4].Data))
-				r.Equal(getHealthString(service[compName].Healthy), fmt.Sprintf("%s", row.Cells[5].Data))
-				r.Equal(service[compName].Message, fmt.Sprintf("%s", row.Cells[6].Data))
+			tb, err := buildApplicationListTable(ctx, client, tc.namespace)
+			r.Equal(tc.expectedErr, err)
+			for _, app := range tc.apps {
+				for i, component := range app.Spec.Components {
+					row := tb.Rows[i+1]
+					compName := fmt.Sprintf("%s", row.Cells[1].Data)
+					r.Equal(component.Name, compName)
+					r.Equal(component.Type, fmt.Sprintf("%s", row.Cells[2].Data))
+					r.Equal(string(app.Status.Phase), fmt.Sprintf("%s", row.Cells[4].Data))
+					r.Equal(getHealthString(service[compName].Healthy), fmt.Sprintf("%s", row.Cells[5].Data))
+					r.Equal(service[compName].Message, fmt.Sprintf("%s", row.Cells[6].Data))
+				}
+			}
+			if len(tc.labelSelector) > 0 {
+				appName := fmt.Sprintf("%s", tb.Rows[1].Cells[0].Data)
+				r.Equal(len(tc.apps)-1, len(tb.Rows)-1)
+				r.Equal(tc.apps[1].Name, appName)
 			}
```

Comment:
This snippet works only in the case:
1. test case have x app and (x-1) app have label.
2. second app of test case is labeled and listed.

Please consider making the test more robust, for example add `expectedAppList int` field 
 to testCases.

Reply:
Done